### ISSUE TITLE
createTodo action after allAreDone action shows the cleared data again.

### ISSUE THREAD

**Issue by rahulcs** [emberjs/quickstart-code-sample#1]
Issue was found in: http://jsbin.com/aqexej/3/edit

Steps to reproduce issue:
1. Add 2 items to the todo list.
2. 'Mark all as complete'.
3. Add another item to the todo.
4. It shows all the cleared items also. 

Your example works fine for ember-1.0.0-rc.5.min.js, handlebars-1.0.0-rc.4.js, but you need to change the ember-data file to the latest.

> 259760 B  2013-06-01T18:35:28.000Z  ember-data-latest.js

It seemed to work fine with June 1st build of ember data. 
![before](https://f.cloud.github.com/assets/624925/605542/fc43b4be-cd18-11e2-992d-3ec10bbe4dc6.png)

I cleared everything which was completed. Yet, in the next image it came back in the list of things. This issue only happens on the index route.

![after](https://f.cloud.github.com/assets/624925/605543/03d97f7e-cd19-11e2-98ca-f2b6ff4d874a.png)

Basically using the latest ember-data, this issue does not appear.
